compiler/modules/front.syntax/parse.bal

```diff
@@ -185,11 +185,13 @@ function parseFunctionDefinition(Tokenizer tok, ModulePart part, Visibility vis,
     check tok.advance();
     Position namePos = tok.currentStartPos();
     string name = check tok.expectIdentifier();
-    FunctionParam [] params = [];
+    FunctionParam[] params = [];
     FunctionTypeDesc typeDesc = check parseFunctionTypeDesc(tok, params);
+    // if restParam is not nil it must have a name
+    FunctionParam? restParam = <FunctionParam?>typeDesc.restParam;
```

Comment:
This is not good.  parseFunctionTypeDesc handles this neatly (i.e. without casts) for non rest params.

One approach is for parseFunctionTypeDesc to include the rest param in the params list.

Reply:
fixed (no longer required)